### ISSUE TITLE
Pytorch Model Compression on Object detection model using AIMET compression API's

### ISSUE THREAD

**Issue by navaravan** [quic/aimet#1875]
Hi @quic-akhobare

I have a object detection model in Pytorch, I want to use AIMET pytorch Model compression API's on it, to see how much reduction can be achieved by it.
Currently in the examples I can see  classifier Resnet50 being used for Model Compressions.
So I want to know what changes (data loading etc.) I have to do in the code, so that i can compress my pytorch model

**Comment by quic-akhobare** [comment]
Tagging: @quic-hitameht and @quic-mangal 

Hi @navaravan - in theory you can just use the Resnet50 example and modify that for your object-detect model. Please try that and let us know how it goes.

**Comment by padeirocarlos** [comment]
Hi @quic-akhobare,

could you give more details about this idea for object detection models? for example I have a object detection model in Pytorch a faster-RCNN with resnet-50 backbone.

Case 1: I already trined faster-RCNN with resnet-50 and l have mAP 0.56 accuracy and I want to prune the model. I can prune the backbone then fine-tune the entire faster-RCNN? or

Case 2: I need to prune reset-50 before use it into faster-50? then train faster-RCNN with Resent-50 pruned?

**Comment by quic-akhobare** [comment]
Hi @padeirocarlos - your explanation was a bit confusing to me. But let me say this - you can prune just the backbone and then fine-tune backbone + head. I think that would be preferable approach. Let me know if that makes sense.

**Comment by padeirocarlos** [comment]
Hi @quic-akhobare thank you for your feedback. I managed to prune the backbone. I am fining-tune the model because the accuracy dropped. But I would like to know , How can I measure the memory footprint imptoviment? Is there any way?

**Comment by quic-mangal** [comment]
@padeirocarlos the MAC or memory cost which are printed as part of statistics should have reduced. We also print the total compression ratio with which the model was compressed.
